tests/ui/traits/next-solver/normalize-unsize-rhs.rs

```diff
@@ -1,5 +1,6 @@
 // compile-flags: -Znext-solver
-// check-pass
```

Comment:
have to figure out why this fails now :/ do we not `select_where_possible` at some important point? probably

Reply:
Why would this turn into `non-primitive cast` error due to a missing `select_where_possible`?